### PR TITLE
Plugins: Improve c plugin (doc, tests, refactor)

### PR DESCRIPTION
**Pull request by flo91**
- some small refactoring (e.g. remove unnecessary dups, reduce scope of variables)
- add code documentation (Doxygen)
- extended README.md
- add unit tests

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which contains `_(my name)_`)
      **Please always add them to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - add a line in `doc/news/_preparation_next_release.md`
  - reformat the code with `scripts/dev/reformat-all`
  - make all unit tests pass
  - fix all memleaks
  - fix the CI itself (or rebase if already fixed)
- [x] The PR is rebased with current master.

## Checklist

- [x] I added unit tests for my code
- [x] I fully described what my PR does in the documentation
      (not in the PR description)
- [x] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [x] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [x] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if everything is done** and no further pushes are planned by you.

### CODE REVIEW

**Comment by atmaxinger** [comment]
In what state is this? Do you think this should also be part of the v0.11 release?

**Comment by flo91** [comment]
This PR is basically finished and ready for reviews.
If the reviewers approve the changes, I don't see a reason why we shouldn't include it in v0.11.

File: src/plugins/c/README.md [hunk 1]
```diff
@@ -6,15 +6,56 @@
...
+kdb set user:/tests/cplugin/key2 value2
+#> Create a new key user:/tests/cplugin/key2 with string "value2"
+
+kdb meta-set user:/tests/cplugin/key2 metakey2.1 metaval2.1
+kdb set user:/tests/cplugin/key2 metakey2.2 metaval2.2
+
+kdb set user:/tests/cplugin/key3 value3
+#> Create a new key user:/tests/cplugin/key3 with string "value3"
+
+bin/kdb export user:/tests/cplugin c
```
Comment (markus2330) [inline src/plugins/c/README.md:55]:
```suggestion
kdb export user:/tests/cplugin c
```

File: src/plugins/c/README.md [hunk 2]
```diff
@@ -6,15 +6,56 @@
...
+Export a subset of the KDB as C code which creates a KeySet (`ksNew(<number of keys>, keyNew(...), ...)`).
+This can be useful for generating test data or help to generate code for already defined configuration data.
 
 ```sh
 kdb export user:/testdata c
 ```
+
+The output can be used in C code for generating KeySets and the keys inside them.
+So one use case is to extract a part of the KDB and copy the generated code
+to create the exported part of the KDB programmatically, e.g. for another Elektra installation.
```
Comment (markus2330) [inline src/plugins/c/README.md:24]: one line per sentence according to documentation guidelines.

File: src/plugins/c/c.c [hunk 3]
```diff
@@ -150,58 +172,75 @@ int keyGenerate (const Key * key, FILE * stream)
...
 
-	ksDel (cks);
 	return 1;
 }
 
-int elektraCGet (Plugin * handle ELEKTRA_UNUSED, KeySet * returned ELEKTRA_UNUSED, Key * parentKey ELEKTRA_UNUSED)
+/**
+ * @brief The get function of this plugin is only used to define the contract.
+ *
+ * @return 1
```
Comment (markus2330) [inline src/plugins/c/c.c:208]:
```suggestion
 * @retval 1 on success (always as parsing is not yet implemented)
```

**Review by markus2330 — APPROVED** [review]
Great, yes, let us take this in. Shell recorder tests would have been nice.

**Comment by atmaxinger** [comment]
Great work!

**Comment by atmaxinger** [comment]
jenkins build libelektra please

**Comment by atmaxinger** [comment]
jenkins build libelektra please

**Comment by atmaxinger** [comment]
jenkins build libelektra please

**Pull request merged**